browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchClient.java

```diff
@@ -107,6 +107,42 @@ private boolean isValidRequestCode(int requestCode) {
      * @param activity the activity that received the deep link back into the app
      */
     public BrowserSwitchResult deliverResult(@NonNull FragmentActivity activity) {
+        BrowserSwitchResult result = getResult(activity);
+        if (result != null) {
+
+            Context appContext = activity.getApplicationContext();
+            BrowserSwitchRequest request = persistentStore.getActiveRequest(appContext);
+
+            @BrowserSwitchStatus int status = result.getStatus();
+            switch (status) {
+                case BrowserSwitchStatus.SUCCESS:
+                    // ensure that success result is delivered exactly once
+                    persistentStore.clearActiveRequest(appContext);
+                    break;
+                case BrowserSwitchStatus.CANCELED:
+                    // ensure that cancellation result is delivered exactly once, but allow for
+                    // a cancellation result to remain in shared storage in case it
+                    // later becomes successful
+                    request.setShouldNotifyCancellation(false);
+                    persistentStore.putActiveRequest(request, activity);
+                    break;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Peek at a pending browser switch result to an Android activity.
+     *
+     * We recommend you call this method in onResume to receive a browser switch result once your
+     * app has re-entered the foreground.
+     *
+     * This can be used in place of {@link BrowserSwitchClient#deliverResult(FragmentActivity)} when
```

Comment:
I'm confused how this would work in a merchant app - they should call `getResult` and then also `deliverResult`? I'd opt to keep this method package-private unless we really think merchants need it because it introduces room for integration error in my opinion. If we need a CHANGELOG entry for the release we could just say `Added internal methods for Braintree Android Drop-in` or something.

Reply:
Yeah I'm up for that. The CHANGELOG would've been my only reservation.